query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/simple/find_unique_or_throw.rs

```diff
@@ -0,0 +1,81 @@
+use query_engine_tests::*;
+
+#[test_suite(schema(schemas::user))]
+mod find_unique_or_throw {
+    use query_engine_tests::assert_query;
+
+    #[connector_test]
+    async fn find_unique_or_throw_when_record_is_found(runner: Runner) -> TestResult<()> {
+        test_user(&runner).await?;
+
+        assert_query!(
+            &runner,
+            r#"query { findUniqueOrThrowUser(where: { email: "a@b.com" }) { id } }"#,
+            r#"{"data":{"findUniqueOrThrowUser":{"id":1}}}"#
+        );
+
+        assert_query!(
+            &runner,
+            r#"query { findUniqueOrThrowUser(where: { first_name_last_name: { first_name: "Elongated", last_name: "Muskrat" } }) { id } }"#,
+            r#"{"data":{"findUniqueOrThrowUser":{"id":1}}}"#
+        );
+
+        assert_query!(
+            runner,
+            "query { findUniqueOrThrowUser(where: { id: 1 }) { id } }",
+            r#"{"data":{"findUniqueOrThrowUser":{"id":1}}}"#
+        );
+
+        Ok(())
+    }
+
+    #[connector_test]
+    async fn no_result_find_unique_by_id(runner: Runner) -> TestResult<()> {
+        test_user(&runner).await?;
+
+        assert_error!(
+            &runner,
+            "query { findUniqueOrThrowUser(where: { id: 2 }) { id } }",
+            2025,
+            "An operation failed because it depends on one or more records that were required but not found. Expected a record, found none."
```

Comment:
This is the error returned to the user, which as per [this method](https://github.com/prisma/prisma-engines/pull/3356/files#diff-82cd668bcd639292fedb193dacba5d69d40575e5735b25bac6c4896652ddfcf7R291-R304) is a ConnectorError rendering [this](https://www.prisma.io/docs/reference/api-reference/error-reference#p2025)